staging/src/k8s.io/apiserver/pkg/storage/watch_cache.go

```diff
@@ -203,21 +208,13 @@ func objectToVersionedRuntimeObject(obj interface{}) (runtime.Object, uint64, er
 	if err != nil {
 		return nil, 0, err
 	}
-	resourceVersion, err := parseResourceVersion(meta.GetResourceVersion())
+	resourceVersion, err := w.versioner.ParseResourceVersion(meta.GetResourceVersion())
```

Comment:
I think you could use resourceVersion, err := w.versioner.ObjectResourceVersion(object) and get rid of lines 207-211

The ParseResourceVersion function does something differently with the error returned from what the function you removed (parseResourceVersion) would return.

Reply:
Then you could also remove the "k8s.io/apimachinery/pkg/api/meta" import